packages/xstate-graph/src/types.ts

```diff
@@ -57,72 +56,112 @@ export type DirectedGraphNode = JSONSerializable<
+      state: TState;
       event: TEvent;
     }
   >;
 }
 
-export interface StatePaths<TContext, TEvent extends EventObject> {
+export interface StatePlan<TState, TEvent extends EventObject> {
   /**
    * The target state.
    */
-  state: State<TContext, TEvent>;
+  state: TState;
   /**
    * The paths that reach the target state.
    */
-  paths: Array<StatePath<TContext, TEvent>>;
+  paths: Array<StatePath<TState, TEvent>>;
 }
 
-export interface StatePath<TContext, TEvent extends EventObject> {
+export interface StatePath<TState, TEvent extends EventObject> {
   /**
    * The ending state of the path.
    */
-  state: State<TContext, TEvent>;
+  state: TState;
   /**
-   * The ordered array of state-event pairs (segments) which reach the ending `state`.
+   * The ordered array of state-event pairs (steps) which reach the ending `state`.
    */
-  segments: Segments<TContext, TEvent>;
+  steps: Steps<TState, TEvent>;
   /**
-   * The combined weight of all segments in the path.
+   * The combined weight of all steps in the path.
    */
   weight: number;
 }
 
-export interface StatePathsMap<TContext, TEvent extends EventObject> {
-  [key: string]: StatePaths<TContext, TEvent>;
+export interface StatePlanMap<TState, TEvent extends EventObject> {
+  [key: string]: StatePlan<TState, TEvent>;
 }
-export interface Segment<TContext, TEvent extends EventObject> {
+
+export interface Step<TState, TEvent extends EventObject> {
   /**
    * The current state before taking the event.
    */
-  state: State<TContext, TEvent>;
+  state: TState;
   /**
    * The event to be taken from the specified state.
    */
   event: TEvent;
 }
 
-export type Segments<TContext, TEvent extends EventObject> = Array<
-  Segment<TContext, TEvent>
+export type Steps<TState, TEvent extends EventObject> = Array<
+  Step<TState, TEvent>
 >;
 
 export type ExtractEvent<
   TEvent extends EventObject,
   TType extends TEvent['type']
 > = TEvent extends { type: TType } ? TEvent : never;
 
-export interface ValueAdjMapOptions<TContext, TEvent extends EventObject> {
+export interface ValueAdjMapOptions<TState, TEvent extends EventObject> {
   events?: {
     [K in TEvent['type']]?:
       | Array<ExtractEvent<TEvent, K>>
-      | ((state: State<TContext, TEvent>) => Array<ExtractEvent<TEvent, K>>);
+      | ((state: TState) => Array<ExtractEvent<TEvent, K>>);
   };
-  filter?: (state: State<TContext, any>) => boolean;
-  stateSerializer?: (state: State<TContext, any>) => string;
+  filter?: (state: TState) => boolean;
+  stateSerializer?: (state: TState) => string;
   eventSerializer?: (event: TEvent) => string;
 }
+
+export interface VisitedContext<TState, TEvent> {
+  vertices: Set<SerializedState>;
+  edges: Set<SerializedEvent>;
+  a?: TState | TEvent; // TODO: remove
+}
+
+export interface SerializationOptions<TState, TEvent> {
+  serializeState: (state: TState, event: TEvent | null) => SerializedState;
+  serializeEvent: (event: TEvent) => SerializedEvent;
```

Comment:
If I read things correctly then those options are public. Requiring people to provide functions that return those branded types feels like a chore on their side - it won't be super obvious and we'd have to document it. I don't think it's worth it. We should expect a simple `string`s here and cast those functions internally if we want to use those branded types internally for improved type safety

Reply:
Added as a task above.